testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksDelete.java

```diff
@@ -53,4 +58,48 @@ public void testDeleteOnAppendOnlyTableFails()
                 .containsOnly(row(1, 11), row(2, 12));
         onTrino().executeQuery("DROP TABLE " + tableName);
     }
+
+    // Databricks 12.1 added support for deletion vectors
+    // TODO: Add DELTA_LAKE_OSS group once after upgrading OSS Delta Lake image to 2.3.0
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, DELTA_LAKE_EXCLUDE_113, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testDeletionVectors()
+    {
+        String tableName = "test_deletion_vectors_" + randomNameSuffix();
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "         (a INT, b INT)" +
+                "         USING delta " +
+                "         LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' " +
+                "         TBLPROPERTIES ('delta.enableDeletionVectors' = true)");
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1,11), (2, 22)");
+            onDelta().executeQuery("DELETE FROM default." + tableName + " WHERE a = 2");
+
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
+                    .containsOnly(row(1, 11));
+            // TODO https://github.com/trinodb/trino/issues/16903 Add support for deletionVectors reader features
+            assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                    .hasMessageContaining("Unsupported reader features: [deletionVectors]");
+
+            assertThat(onTrino().executeQuery("SHOW COLUMNS FROM delta.default." + tableName).project(1))
+                    .containsOnly(row("a"), row("b"));
+
+            assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (3, 33)"))
+                    .hasMessageContaining("Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + tableName))
+                    .hasMessageContaining("Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("UPDATE delta.default." + tableName + " SET a = 3"))
+                    .hasMessageContaining("Delta Lake writer version 7 which is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO delta.default." + tableName + " t USING delta.default." + tableName + " s " +
+                    "ON (t.a = s.a) WHEN MATCHED THEN UPDATE SET b = -1"))
+                    .hasMessageContaining("Delta Lake writer version 7 which is not supported");
+
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
```

Comment:
Duplicated query   - see https://github.com/trinodb/trino/pull/16905/commits/50c321e4a9ca708e8582c9cead2c73fa8413ded0#diff-b76aa3b1811ce8db0a40df6da6d3a1a90fd581c98f745436cc85700bb91186b3R79

Reply:
That's intentional to ensure Trino DML statements didn't change rows.